main.tf

```diff
@@ -1,227 +1,54 @@
-}
-
-module "exec_label" {
-  source      = "cloudposse/label/null"
-  version     = "0.25.0"
-  attributes  = ["exec"]
-  label_order = var.label_orders.iam
-
-  context = module.this.context
-}
-
-data "aws_iam_policy_document" "ecs_task_exec" {
-  statement {
-    actions = ["sts:AssumeRole"]
-
-    principals {
-      type        = "Service"
-      identifiers = ["ecs-tasks.amazonaws.com"]
-    }
-  }
-}
-
-resource "aws_iam_role" "ecs_exec" {
-  count = var.redis_enabled ? 1 : 0
-
-  name               = module.exec_label.id
-  assume_role_policy = data.aws_iam_policy_document.ecs_task_exec.json
-  tags               = module.exec_label.tags
-}
-
-data "aws_iam_policy_document" "ecs_exec" {
-  statement {
-    effect    = "Allow"
-    resources = ["*"]
-
-    actions = [
-      "ssm:GetParameters",
-      "ecr:GetAuthorizationToken",
-      "ecr:BatchCheckLayerAvailability",
-      "ecr:GetDownloadUrlForLayer",
-      "ecr:BatchGetImage",
-      "logs:CreateLogGroup",
-      "logs:CreateLogStream",
-      "logs:PutLogEvents"
-    ]
-  }
-}
-
-resource "aws_iam_role_policy" "ecs_exec" {
-  count = var.redis_enabled ? 1 : 0
-
-  name   = module.exec_label.id
-  policy = data.aws_iam_policy_document.ecs_exec.json
-  role   = aws_iam_role.ecs_exec[0].id
-}
-
-module "container_definition" {
-  count   = var.redis_enabled ? 1 : 0
-  source  = "cloudposse/ecs-container-definition/aws"
-  version = "0.58.1"
-
-  container_image              = "${var.redis_image_repository}:${var.redis_image_tag}"
-  container_name               = local.redis_container_name
-  container_cpu                = var.redis_cpu_size
-  container_memory_reservation = var.redis_service_memory_size
-
-  port_mappings = [
+module "ddb" {
+  count   = var.ddb_enabled ? 1 : 0
+  source  = "terraform-aws-modules/dynamodb-table/aws"
+  version = "3.2.0"
+
+  name                = module.kvstore_label.id
+  hash_key            = "key"
+  tags                = module.kvstore_label.tags
+  autoscaling_enabled = var.ddb_autoscaling_enabled
+  billing_mode        = var.ddb_billing_mode
+  read_capacity       = local.default_read_capacity
+  write_capacity      = local.default_write_capacity
+
+  attributes = [
     {
-      containerPort = 6379
-      hostPort      = 0
-      protocol      = "tcp"
-    },
-  ]
-
-  command = [
-    "--maxmemory ${var.redis_memory_size}mb",
-    "--maxmemory-policy allkeys-lru"
+      name = "key"
+      type = "S"
+    }
   ]
 }
 
 module "redis" {
-  count   = var.redis_enabled ? 1 : 0
-  source  = "cloudposse/ecs-alb-service-task/aws"
-  version = "0.66.4"
-
-  context = module.redis_label[0].context
-
-  container_definition_json          = "[${sensitive(module.container_definition[0].json_map_encoded)}]"
+  count  = var.redis_enabled ? 1 : 0
+  source = "github.com/justtrackio/terraform-aws-ecs-redis?ref=add_service_discovery_name_varibale"
+
+  context      = module.kvstore_label.context
+  label_orders = var.label_orders
```

Comment:
assuming we have a label order of `["environment","stage","name"]`
wouldn't that result in this dynamodb table name `env-stg-name`, while the redis' ecs service is going to be also called like that?
In my opinion, this doesn't make sense, when we have a cluster name of `env`.
Shouldn't we revert to the label order per service again here?

Reply:
It uses the label_orders ecs not kvstore. I put it inside this module as well so that we can override it if we want.